python/ray/exceptions.py

```diff
@@ -247,11 +262,32 @@ def get_creation_task_error(self):
         return None
 
     def __str__(self):
-        if self.creation_task_error:
-            return ("The actor died because of an error" +
-                    " raised in its creation task, " +
-                    self.creation_task_error.__str__())
-        return ("The actor died unexpectedly before finishing this task.")
+        return self.error_msg
+
+    def _gen_error_msg_from_cause(self, cause: ActorDeathCause):
+        """Generate the error message from the
+        actor died error cause protobuf message.
+        """
+        if not cause:
+            return (f"{self.base_error_msg} due to an unknown reason.")
+
+        if cause.HasField("worker_died_context"):
+            return (f"{self.base_error_msg} "
+                    f"{cause.worker_died_context.error_message}")
+        elif cause.HasField("node_died_context"):
+            return (f"{self.base_error_msg} "
+                    f"{cause.node_died_context.error_message}")
+        elif cause.HasField("owner_died_context"):
+            return (f"{self.base_error_msg} "
+                    f"{cause.owner_died_context.error_message}")
+        elif cause.HasField("killed_by_app_context"):
+            return (f"{self.base_error_msg} "
+                    f"{cause.killed_by_app_context.error_message}")
+        elif cause.HasField("out_of_scope_context"):
+            return (f"{self.base_error_msg} "
+                    f"{cause.out_of_scope_context.error_message}")
+        else:
+            assert False, f"Unknown death cause. {cause}"
```

Comment:
This still looks like unnecessary complexity here. Can we use a generalized schema so you can write:

```return "The actor died: {self.base_error_msg}: {self.cause}"?```

Reply:
So, I'd prefer this to be in a separate PR. What about I make a PR to generalize the schema (I am only going to have error_message and construct all metadata into it), merge that, and then we come back to this PR after?